src/components/Presets/AddPreset.svelte

```diff
@@ -0,0 +1,76 @@
+<script lang="ts">
+  import { version } from '../../../package.json'
+  import { patchStore, presetsStore } from '../../stores'
+  import { type Patch, Visibility } from '$lib/patch'
+  import { savePreset } from '$lib/presets'
+  import { DEFAULT_CATEGORIES } from '$lib/presets/constants'
+
+  export let handleCancelClick: () => void
+
+  const handleSubmit = async (event: Event): Promise<void> => {
+    const formEl = event.target as HTMLFormElement
+    const data: FormData = new FormData(formEl)
+
+    const preset: Patch = {
+      creator: data.get('creator') as string,
+      favorite: false,
+      id: crypto.randomUUID(),
+      name: data.get('name') as string,
+      notes: data.get('notes') as string,
+      params: {
+        delayTime: 200,
+        feedback: 0,
+        mix: 50
+      },
+      // @ts-ignore: Type mismatch from form data
+      tags: (data.get('tags') ? data.get('tags')?.split(',')?.map((tag: string) => tag.trim()) : []) as string[],
```

Comment:
I may revisit these. I was having a hard time getting the form data types to play well with the `Patch` type

Reply:
One way to get better types here would be to parse the data into an object, then write a type guard to validate that the object is a `Patch`. If it is, assign it to `preset`.

Here's an example type guard from the branch I am working on:

https://github.com/webnative-examples/ditto/blob/a2476478a88381be0c9dfa018a6e3758e412d3af/src/lib/auth/connected.ts#L4-L14

Essentially we are saying, this is an `unknown` thing, but we are going to parse it to make sure it has the things it should. When we are done, you can trust us TypeScript. It's the right thing.

We use this type guard like this:

https://github.com/webnative-examples/ditto/blob/a2476478a88381be0c9dfa018a6e3758e412d3af/src/lib/auth/connected.ts#L32-L37

On line 33, we know for sure that we are working with a `ConnectedStatus`, and we can safely access the `connected` property.

The advantage of a type guard over `any` is that we can have confidence about the type going forward. Using `any` turns off type checking and can lead to situations where we misuse a variable and TypeScript can't help us.

Reply:
Nice, i'll give that a go, thanks 👌🏼

Reply:
This was getting a little involved due in part to `tags` needing to be converted from an array to a string then back again. I ended up going the route of binding values to the form inputs and doing away with using `FormData` for now. Certainly not the best solution, but it seems to do the trick for now. Happy to revisit this though